### ISSUE TITLE
no-unused-modules false positives for files imported using a typescript baseUrl even with typescript parser

### ISSUE THREAD

**Issue by tomdglenn91** [import-js/eslint-plugin-import#2404]
I have an application generated with create-react-app. Inside this app is a folder called `components` with two files, Test and TestTwo. Inside App.tsx, i import from these in relative and absolute ways, using the baseUrl setting in tsconfig. 

tsconfig:
`baseUrl: ./src`

App.tsx:
```
import Test from './components/Test'
import TestTwo from 'components/TestTwo'
```

`Test` has no issues, but `TestTwo` is flagged as an unused export despite being imported. 

I have searched around the web and the only answers I found were relating to the typescript resolver as per your docs, but these have been followed.

**eslintrc.json**
```
const path = require('path')

module.exports = {
	extends: [
		'eslint:recommended',
		'plugin:react/recommended',
		'plugin:react/jsx-runtime',
		'react-app',
		'plugin:import/typescript'
	],
	settings: {
		'import/resolver': {
			typescript: {},
		},
	},
	rules: {
		'import/no-unused-modules': [
			'warn',
			{
				unusedExports: true,
				missingExports: true,
				src: [path.join(__dirname, 'src')],
			},
		],
	}
}
```

**tsconfig.json**
```
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",
    "baseUrl": "./src"
  },
  "include": [
    "src"
  ]
}
```

I've uploaded a repo here: https://github.com/tomdglenn91/basepath-unused-exports

I would expect that files imported via absolute paths using the baseUrl would not flag as unused. I can only assume that there is something wrong with the way I have configured it?

**Comment by tomdglenn91** [comment]
In true issue-raising fashion, despite checking my own work for 2 days, after posting the issue, I discover the cause of my troubles. Turns out it was the `node/paths` i had in an extensions that was messing with it. Not sure how it worked but removing fixed my demo. I'm still seeing issues but I think they're unrelated and likely my own config. Either way not related to baseUrl so closing this.
